python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -472,6 +477,19 @@ def stream_raw_events(self) -> Iterator[DbtCliEventMessage]:
                 if is_error_message:
                     self._error_messages.append(str(event))
 
+                # Attempt to parse the columns metadata from the event message.
+                # If it exists, then save it as metadata for the dbt node in the manifest.
```

Comment:
I would lean away from mutating the manifest. A couple reasons:
- Conceptually, this isn't part of the manifest. I.e. the manifest refers specifically to the output of specific dbt commands, and this muddies that definition.
- Practically, it seems like this could cause confusing/unexpected/buggy behavior; the manifest used in one place in the program gets mysteriously mutated in a different part of the program, and it's difficult to trace that provenance.

Reply:
Fair! I'll add it as a separate field on `DbtCliEventMessage`.